skyVegasTests/gamePlayGameClient.spec.ts

```diff
@@ -1,35 +1,261 @@
+                await page
+                    .frameLocator('#root iframe')
+                    .locator('em.value.main-wallet')
+                    .textContent()
+            );
+
+            const stakeAmountConsole = await getValueFromConsoleMessages(
+                consoleMessages,
+                ExpectedMessage.STAKE_UPDATE
+            );
+
+            const stakeAmountScreen = currencyStringToNumber(
+                await page
+                    .frameLocator('#root iframe')
+                    .locator('em.value.stake-wallet')
+                    .textContent()
+            );
+
+            await firstSpin(page, consoleMessages);
+            numberOfSpins++;
+
+            let totalWinAmountConsole = 0;
+            let winAmountConsole = await getValueFromConsoleMessages(
+                consoleMessages,
+                ExpectedMessage.WIN_UPDATE
+            );
+            totalWinAmountConsole += winAmountConsole;
+
+            let totalWinAmountScreen = 0;
+            let winAmountScreen = currencyStringToNumber(
+                await page
+                    .frameLocator('#root iframe')
+                    .locator('em.value.won-wallet')
+                    .textContent()
+            );
+            totalWinAmountScreen += winAmountScreen;
+
+            while (winAmountConsole > 0 && winAmountScreen > 0) {
+                await spin(page, consoleMessages);
+                numberOfSpins++;
+
+                winAmountConsole = await getValueFromConsoleMessages(
+                    consoleMessages,
+                    ExpectedMessage.WIN_UPDATE
+                );
+                totalWinAmountConsole += winAmountConsole;
+
+                winAmountScreen = currencyStringToNumber(
+                    await page
+                        .frameLocator('#root iframe')
+                        .locator('em.value.won-wallet')
+                        .textContent()
+                );
+                totalWinAmountScreen += winAmountScreen;
+            }
+
+            const endBalanceConsole = await getValueFromConsoleMessages(
+                consoleMessages,
+                ExpectedMessage.BALANCE_UPDATE
+            );
+
+            const endBalanceScreen = currencyStringToNumber(
+                await page
+                    .frameLocator('#root iframe')
+                    .locator('em.value.main-wallet')
+                    .textContent()
+            );
+
+            const endWinLossAmountScreen = currencyStringToNumber(
+                await page
+                    .frameLocator('#root iframe')
+                    .locator('em.value.gaming-session-profit-and-loss')
+                    .textContent()
+            );
+
+            const endWinLossAmountConsole = Number(
+                (
+                    totalWinAmountConsole -
+                    stakeAmountConsole * numberOfSpins
+                ).toFixed(2)
```

Comment:
Rather than the Number(...).toFixed(2) how about a "toCurrency" helper function that shows a bit clearer what this is doing

Reply:
Will make this change in the PR for GTX-441. We had to pull that work into this one as there were overlaps with helper functions that were needed.